### PR TITLE
Fix the returned position from `Meta.parse` in non-greedy mode

### LINKED ISSUE #32958: Meta.parse consumes inconsistent amount of input in non-greedy mode
`Meta.parse` in non-greedy mode uses the character stream position (via `io.pos` in `jl-parse-one`), but the parser may already have read the next token using `peek-token`. This can cause the returned position to be inconsistent. For example:

```julia
julia> Meta.parse("x,", 1, greedy=false)
(:x, 2)

julia> Meta.parse(":,", 1, greedy=false)
(:(:), 3)  # should this be (:(:), 2) ?
```

I found this problem while chasing down a bug in `Base.shell_parse` caused by #32408.

### PR DESCRIPTION
**Pull request by ghost**
Fixes #32958.

### CODE REVIEW

File: src/julia-parser.scm [hunk 1]
```diff
@@ -558,7 +558,7 @@
 
 ;; --- token stream ---
 
-(define (make-token-stream s) (vector #f s #t #f #f))
+(define (make-token-stream s) (vector #f s #t #f #f (io.pos s) #f))
```
Comment (JeffBezanson) [inline src/julia-parser.scm]: Please add accessors and setters for these like `ts:port` below.

File: src/julia-parser.scm [hunk 2]
```diff
@@ -2465,5 +2473,5 @@
                (begin (take-token s) (skip-loop (peek-token s)))))
          (if (eof-object? (peek-token s))
              (eof-object)
-             ((if (null? production) parse-stmts (car production))
-              s)))))
+             (let ((res ((if (null? production) parse-stmts (car production))
+              s))) (io.seek (ts:port s) (aref s 5)) res)))))
```
Comment (JeffBezanson) [inline src/julia-parser.scm]: Confusing spacing here.

**Comment by mikan2087** [comment]
Thank you for reviewing. I added commits to fix.

**Comment by DilumAluthge** [comment]
It looks like this pull request is no longer mergeable.

If you would still like to make this change, please make a new pull request based on the latest `master` branch of Julia.

**Comment by vtjnash** [comment]
@JeffBezanson do you want to take over and land this?

**Pull request closed without merging**